opensrp-reporting/src/main/java/org/smartregister/reporting/repository/DailyIndicatorCountRepository.java

```diff
@@ -317,7 +422,105 @@ private void extractIndicatorTalliesFromMultiResult(@NonNull Map<String, Indicat
         return tallyMap;
     }
 
+    public Map<String, List<IndicatorTally>> findTalliesInMonth(@NonNull Date month) {
+        SimpleDateFormat dateFormat = new SimpleDateFormat(ReportIndicatorDaoImpl.DAILY_TALLY_DATE_FORMAT, Locale.getDefault());
+        Map<String, List<IndicatorTally>> talliesFromMonth = new HashMap<>();
+        Cursor cursor = null;
+        try {
+            Calendar startDate = Calendar.getInstance();
+            startDate.setTime(month);
+            startDate.set(Calendar.DAY_OF_MONTH, 1);
+            startDate.set(Calendar.HOUR_OF_DAY, 0);
+            startDate.set(Calendar.MINUTE, 0);
+            startDate.set(Calendar.SECOND, 0);
+            startDate.set(Calendar.MILLISECOND, 0);
+
+            Calendar endDate = Calendar.getInstance();
+            endDate.setTime(month);
+            endDate.add(Calendar.MONTH, 1);
+            endDate.set(Calendar.DAY_OF_MONTH, 1);
+            endDate.set(Calendar.HOUR_OF_DAY, 23);
+            endDate.set(Calendar.MINUTE, 59);
+            endDate.set(Calendar.SECOND, 59);
+            endDate.set(Calendar.MILLISECOND, 999);
+            endDate.add(Calendar.DATE, -1);
```

Comment:
Wouldn't the end date be hour, minute and second 0? :thinking:

Reply:
The end is 1 millisecond before the hour 0 at min, sec, milisec = 0. And the query includes a `<=` to make that time inclusive. What do you think?

Reply:
Yeah, what I was suggesting was not taking into account `endDate.add(Calendar.DATE, -1);`
So essentially it would be the same as the current implementation